### PR TITLE
Added the query paths

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [x] Documentation Update

## Description

Added the query paths, to get the results based on the query parameters.
ex:
```
http://localhost:3000/optimize_reports?ciSystem=JENKINS&platform=AWS&benchmark=cluster-density-v2&version=4.13&workerCount=20&networkType=OPENSHIFTSDN
```
Screenshot 2023-10-30 at 8 51 46 PM
## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Review by krishvoor — APPROVED** [review]
This is nicely done - very useful to share with others.

File: frontend/src/components/Home/SidePaneView.js [hunk 1]
```diff
@@ -43,6 +50,20 @@ export const SidePaneView = () => {
...
+        let buildParams = ''
+        if(ciSystem !== '') buildParams += `&ciSystem=${ciSystem}`
+        if(platform !== '') buildParams += `&platform=${platform}`
+        if(benchmark !== '') buildParams += `&benchmark=${benchmark}`
+        if(version !== '') buildParams += `&version=${version}`
+        if(workerCount !== '') buildParams += `&workerCount=${workerCount}`
+        if(networkType !== '') buildParams += `&networkType=${networkType}`
+        if(startDate !== '') buildParams += `&startDate=${startDate}`
+        if(endDate !== '') buildParams += `&endDate=${endDate}`
+        history.push(`/optimize_reports?${buildParams.substring(1)}`, { replace: true });
```
Comment (vishnuchalla) [inline frontend/src/components/Home/SidePaneView.js:63]: Why are we replacing the latest entry in the browser history with this new entry?

File: frontend/src/components/Home/SidePaneView.js [hunk 2]
```diff
@@ -43,6 +50,20 @@ export const SidePaneView = () => {
             dispatch(fetchJobsData(startDate, endDate))
     }, [startDate, endDate, dispatch])
 
+    useEffect(() => {
+        let buildParams = ''
+        if(ciSystem !== '') buildParams += `&ciSystem=${ciSystem}`
```
Comment (vishnuchalla) [inline frontend/src/components/Home/SidePaneView.js:55]: Do we need these empty checks for all the vars here? So of them already seems to be having a non-empty value set [here](https://github.com/cloud-bulldozer/ocp-perf-dashboard/blob/main/frontend/src/store/reducers/InitialData.js#L14-L19).

File: frontend/src/components/Home/SidePaneView.js (hunk 1 above)
Comment (athiruma) [reply frontend/src/components/Home/SidePaneView.js:63]: `replace=true` will replace the current URL with a modified URL.

File: frontend/src/components/Home/SidePaneView.js (hunk 2 above)
Comment (athiruma) [reply frontend/src/components/Home/SidePaneView.js:55]: As safety passes an empty value from the URL.

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Review by chentex — APPROVED** [review]
LGTM! Thank you @athiruma!!!

**Pull request merged**